packages/block-editor/src/components/off-canvas-editor/block-contents.js

```diff
@@ -123,7 +123,10 @@ const ListViewBlockContents = forwardRef(
 						} }
 					/>
 				) }
-				<BlockDraggable clientIds={ draggableClientIds }>
+				<BlockDraggable
+					clientIds={ draggableClientIds }
+					dragOrigin="offcanvas"
```

Comment:
As this behaviour is only desired for the Nav block lsit view, I've named the origin as "offcanvas". By default no origin will be applied in other contexts (e.g. main list view) and thus they will behave as per `trunk`.